src/assets/src/hooks/useGoogleAnalytics.ts

```diff
@@ -1,22 +1,72 @@
 import { useState, useEffect } from 'react';
 import GoogleAnalytics from 'react-ga4';
 import { useLocation } from 'react-router-dom';
+import { useOneTrust } from './useOneTrust';
+
 
 export const useGoogleAnalytics = (googleAnalyticsId?: string, debug?: boolean) => {
     let location = useLocation();
+
     const [initialized, setInitialized] = useState(false);
     const [previousPage, setPreviousPage] = useState(null as string | null);
 
     if (googleAnalyticsId && !initialized) {
+        GoogleAnalytics.gtag("consent", "default", {
+            ad_storage: "denied",
+            analytics_storage: "denied",
+            functionality_storage: "denied",
+            personalization_storage: "denied",
+            ad_user_data: "denied",
+            ad_personalization: "denied",
+            wait_for_update: 500
+        });
         setInitialized(true);
         GoogleAnalytics.initialize(googleAnalyticsId, { testMode: debug });
     }
 
+    const [loaded, oneTrustActiveGroups, setOptanonWrapper] = useOneTrust();
+    if (loaded) {
+        console.log("loaded oneTrustActiveGroups" + oneTrustActiveGroups);
+    }
+
     useEffect(() => {
         const page = location.pathname + location.search + location.hash;
         if (googleAnalyticsId && page !== previousPage) {
             setPreviousPage(page);
             GoogleAnalytics.send({ hitType: "pageview", page });
         }
     }, [location]);
+    
+    useEffect(() => {
+        const updateGtagConsent = () => {
+
+            if (oneTrustActiveGroups.includes("C0002")) {
+              GoogleAnalytics.gtag("consent", "update", { analytics_storage: "granted" });
+            }
+            if (oneTrustActiveGroups.includes("C0003")) {
+              GoogleAnalytics.gtag("consent", "update", { functional_storage: "granted" });
+            }
+            if (oneTrustActiveGroups.includes("C0004")) {
+              GoogleAnalytics.gtag("consent", "update", {
+                ad_storage: "granted",
+                ad_user_data: "granted",
+                ad_personalization: "granted",
+                personalization_storage: "granted"
+              });
+            } else {
+              // Remove Google Analytics cookies if tracking is declined
+              document.cookie.split(';').forEach(cookie => {
+                const [name] = cookie.split('=');
+                if (name.trim().match(/^_ga(_.+)?$/)) {
+                  document.cookie = `${name}=;path=/;domain=.${window.location.host.replace(/^(.*\.)?(.+\..+)$/, '$2')};expires=Thu, 01 Jan 1970 00:00:01 GMT`;
```

Comment:
Is this something we have to do or something that's done automatically? Did you find this as a suggestion somewhere?

Reply:
This could be generated dynamically

```
function getSpecificExpirationDate() {
    // Create a new Date object set to January 1, 1970, 00:00:01 GMT
    const date = new Date(Date.UTC(1970, 0, 1, 0, 0, 1));
    // Convert the date to a string in the required format
    return date.toUTCString();
}

document.cookie = `${name}=;path=/;domain=.${window.location.host.replace(/^(.*\.)?(.+\..+)$/, '$2')};expires=expires=${getSpecificExpirationDate}
```

Looks like this is set to expire in past. what is the purpose of setting a date in past? or it is working as cookie to never expire

Reply:
`/^(.*\.)?(.+\..+)$/, '$2'`

to would be helpful to add comment saying what string it is parsing? RegEx is be confusing

Reply:
The code block is provided in https://vpcomm.umich.edu/resources/cookie-disclosure/#3rd-party-google-analytics

Reply:
This is from step 4 of the "GOOGLE ANALYTICS / TAG MANAGER" section on the [UMich cookie disclosure](https://vpcomm.umich.edu/resources/cookie-disclosure/) integration. Seems like denying tracking just deletes the google analytics cookie manually. I'm not a fan of this manual extraction, there might be a way to do it with ga4-react library but I haven't looked into it

Reply:
I've added the use of Cookies.remove() from js-cookie to clear the GA cookies, looks like it works well.